consensus/consortium/v2/consortium.go

```diff
@@ -203,6 +209,89 @@ func (c *Consortium) Author(header *types.Header) (common.Address, error) {
 	return header.Coinbase, nil
 }
 
+// VerifyBlobHeader verifies a block's header blob and corresponding sidecar, returning BlobSideCars
+func (c *Consortium) VerifyBlobHeader(block *types.Block, sidecars []types.BlobTxSidecar) (error, *types.BlobSidecars) {
+	nCommitments := 0
+	for _, sidecar := range sidecars {
+		nCommitments += len(sidecar.Commitments)
+	}
+	if nCommitments > params.MaxBlobsPerBlock {
+		return fmt.Errorf("blobs per block limit exceeded, blobs: %d, limit: %d", nCommitments, params.MaxBlobsPerBlock), nil
+	}
+	header := block.Header()
+	if c.skipBlobCheck(header) {
+		return nil, nil
+	}
+	if err := c.verifyVersionHash(block, sidecars); err != nil {
+		return err, nil
+	}
+	for _, sidecar := range sidecars {
+		if err := c.verifySidecar(sidecar); err != nil {
+			return err, nil
+		}
+	}
+	blobSidecars := types.BlobSidecars{}
+	curIndex := 0
+	for _, tx := range block.Transactions() {
+		if tx.Type() == types.BlobTxType {
+			blobSidecars = append(blobSidecars, &types.BlobSidecar{
+				BlobTxSidecar: sidecars[curIndex],
```

Comment:
need curIndex++ in this if or just use tx.BlobTxSidecar()

Reply:
Oh I forgot to use tx.BlobTxSidecar(), fixed

Reply:
Well actually, there is no sidecars in the block here (and therefore no sidecars in the tx itself), I guess we will just add curIndex++